packages/fiber/src/web/index.tsx

```diff
@@ -78,8 +78,18 @@ function render<TCanvas extends Element>(
 
   if (!fiber) {
     // If no root has been found, make one
+
+    // Create gl
+    const glRenderer = createRendererInstance(gl, canvas)
+
+    // Enable VR if requested
+    if (props.vr) {
+      glRenderer.xr.enabled = true
+      glRenderer.setAnimationLoop((timestamp) => advance(timestamp, true, state))
```

Comment:
I'm not sure what this line does, any insight?

Reply:
I copied it from the older code, see changes

Reply:
See https://threejs.org/docs/#manual/en/introduction/How-to-create-VR-content

Reply:
Cool, that makes sense! Thanks

Reply:
I'm wondering about the `state` argument in the `advance()` call, this should be omitted right? It's not updated and my guess is that the `advance` function requires the latest `state`, which will be fetched automatically if the argument is not present.

Reply:
I think in principle I agree with this. Have you tried building a small demo to test it works?

I have no idea how much work that is, if its lots just say and we can merge and go from there 😄